File: src/angular/hq/src/app/models/quotes/get-quotes-v1.ts
import { ProjectStatus } from '../../enums/project-status';
import { PagedRequestV1 } from '../common/paged-request-v1';
import { PagedResponseV1 } from '../common/paged-response-v1';
import { SortDirection } from '../common/sort-direction';

export interface GetQuotesRequestV1 extends PagedRequestV1 {
  search?: string | null;
  id?: string;
  clientId?: string;
  sortBy: SortColumn;
  sortDirection: SortDirection;
  quoteStatus: ProjectStatus | null;
}

export enum SortColumn {
  QuoteName = 1,
  QuoteNumber = 2,
  ClientName = 3,
  ChargeCode = 4,
  Value = 5,
  Status = 6,
  Date = 7,
}

export interface GetQuotesRecordV1 {
  id: string;
  QuoteNumber: number;
  clientId: string;
  clientName: string;
  name: string;
  quoteNumber: number | null;
  chargeCode: string;
  date: string | null;
  value: number | null;
  status: ProjectStatus;
  hasPDF: boolean;
}

export interface GetQuotesRecordsV1 {
  records: [GetQuotesRecordV1];
  total: number | null;
}

export interface GetQuotesResponseV1
  extends PagedResponseV1<GetQuotesRecordV1> {}

File: src/dotnet/HQ.Abstractions/Quotes/GetQuotesV1.cs


using HQ.Abstractions.Common;
using HQ.Abstractions.Enumerations;


namespace HQ.Abstractions.Quotes
{
    public class GetQuotesV1
    {
        public class Request : PagedRequestV1
        {
            public string? Search { get; set; }
            public Guid? Id { get; set; }
            public Guid? clientId { get; set; }
            public SortColumn SortBy { get; set; } = SortColumn.QuoteName;
            public SortDirection SortDirection { get; set; } = SortDirection.Asc;
            public ProjectStatus? QuoteStatus { get; set; }
        }
        public enum SortColumn
        {
            QuoteName = 1,
            QuoteNumber = 2,
            ClientName = 3,
            ChargeCode = 4,
            Value = 5,
            Status = 6,
            Date = 7,
        }

        public class Response : PagedResponseV1<Record>;
        public class Record
        {
            public Guid Id { get; set; }
            public int QuoteNumber { get; set; }
            public string? ChargeCode { get; set; }
            public Guid ClientId { get; set; }
            public string ClientName { get; set; } = null!;
            public string Name { get; set; } = null!;
            public string? Description { get; set; }
            public DateOnly Date { get; set; }
            public decimal Value { get; set; }
            public ProjectStatus Status { get; set; }
        }
    }
}